css/styles.css

```diff
@@ -1,20 +1,345 @@
+  display: flex;
+  flex-wrap: wrap;
+  max-width: 100%;
+  justify-content: center;
+}
+
+.input__group {
+  display: flex;
+  flex-direction: column;
+}
+
+.input__group label {
+  font-size: 1.2rem;
+  font-weight: bold;
+  margin-bottom: 8px;
+  opacity: 0.7
+}
+
+.form__flex-wrap .input__group label {
+  text-align: center;
+  margin-left: 0px;
+  margin-bottom: 8px;
+  margin-right: 32px;
+}
+
+.input__form i {
+  text-align: center;
+  display: block;
+  font-style: normal;
+  color: #d1d1d1;
+  opacity: 0.9;
+  font-size: 0.9rem;
+  margin: 8px auto;
+}
+
+.input__form>.form__row input {
+  margin: 0px auto;
+}
+
+input {
+  border: 0px solid white;
+  border-radius: 40px;
+  padding: 16px 24px;
+  font-size: 2rem;
+  font-family: inherit;
+  color: inherit;
+  font-weight: bold;
+  transition: 0.2s all;
+  margin: 8px 0px;
+  margin-right: 32px;
+}
+
+input[type=number]:placeholder-shown {
+  background: #f3f3f3;
 }
 
-#turnipTable thead tr:first-child {
-    border: none;
+input[type=number]:not(:placeholder-shown) {
+  background: transparent;
+  color: #0AB5CD;
+}
+
+input[type=number]:focus {
+  outline: none;
+  transform: scale(1.1);
+  background: white;
+  box-shadow: 0 1px 6px rgba(0, 0, 0, 0.04), 0 3px 6px rgba(0, 0, 0, 0.07);
+}
+
+input[type=number]:focus::placeholder {
+  opacity: 0;
+}
+
+input[type=number] {
+  width: 60px;
+  text-align: center;
+}
+
+input::-webkit-outer-spin-button,
+input::-webkit-inner-spin-button {
+  -webkit-appearance: none;
+  margin: 0;
+}
+
+input[type=number] {
+  -moz-appearance: textfield;
+}
+
+.reset-button {
+  font-family: inherit;
+  font-weight: bold;
+  color: #FFBABA;
+  padding: 8px 16px;
+  border-width: 0px;
+  border-radius: 40px;
+  background: transparent;
+  font-size: 1.2rem;
+  transition: 0.2s all;
+  position: relative;
+  margin: 32px auto;
+  opacity: 0.5;
+}
+
+.reset-button:hover {
+  cursor: pointer;
+  opacity: 1;
+  box-shadow: 0 1px 6px rgba(0, 0, 0, 0.04), 0 3px 6px rgba(0, 0, 0, 0.07);
+}
+
+
+.table-wrapper {
+  width: 98%;
+  max-height: 50vh;
+  padding: 16px;
+  margin: 0px auto;
+  box-sizing: border-box;
+  overflow-x: scroll;
+  border-radius: 2px;
+}
+
+#turnipTable {
+  border-collapse: collapse;
+}
+
+#turnipTable span {
+  min-width: 80px;
 }
 
 #turnipTable th {
-    text-align: center;
+  min-width: 80px;
+  font-size: 0.8rem;
 }
 
-#turnipTable td:first-child {
-    white-space: nowrap;
-    text-align: left;
+#turnipTable th div:nth-of-type(1) {
+  margin-bottom: 2px;
+}
+
+#turnipTable th div:nth-of-type(2) {
+  display: flex;
+  justify-content: space-around;
+  opacity: 0.4;
 }
 
 #turnipTable td {
-    text-align: center;
+  min-width: 80px;
+  max-width: 84px;
+  padding: 4px 2px;
+  text-align: center;
+  font-size: 0.8rem;
+  border-right: 1px solid rgba(0, 0, 0, 0.03);
+  border-bottom: 1px solid rgba(0, 0, 0, 0.15);
+}
+
+#turnipTable tbody tr {
+  opacity: 0.6;
+}
+
+#turnipTable tbody tr:hover {
+  cursor: default;
+  opacity: 1;
+}
+
+
```

Comment:
```suggestion
#turnipTable .table-pattern {
  white-space: normal;
}

```

This is a style to still allow the pattern to collapse into multiple lines, to aid with the overall width of the table on narrow displays

Reply:
Although is does reduce the width of the overall table - the table size is such that the patterns would likely end up with a single word on each line, which greatly reduces readability. IMHO this should stay as `nowrap`.

Reply:
That is a good point, and if there is a scroll bar, I guess it doesn't matter too much if there is a bit extra.

Here are the two views, one with and one without, just to give an idea of how they both look.

With
![with wrap](https://user-images.githubusercontent.com/6600325/78800353-ba3cff80-7991-11ea-9697-77d19d1ea105.png)

Without
![without-wrap](https://user-images.githubusercontent.com/6600325/78800358-bd37f000-7991-11ea-8d29-204ef447c3dc.png)

Reply:
Well, I have another concern with the changes to the pattern names. The patterns are named in such a way that they both match the information extracted from the game code, as well as clearly describe the behavior of the pattern itself.

These pattern description changes do not clearly explain how the patterns will behave. I'm not in favor of the renames.

Reply:
I see where you are coming from, I feel like for readability and UX, it doesn't make too much difference as most of the value to a user comes from the prediction data itself and the new names do make it easier to parse. However the loss of matching the game code is a concern.

Reply:
Perhaps a solution that addresses both needs would be to add a legend on the page that maps the short names to their true names?

Reply:
yeah that would work, or a tooltip/hover over effect that has the details, depending on how that would interact with mobile.